GitHubRepositories.ps1

```diff
@@ -2944,6 +2945,593 @@ filter Set-GitHubRepositoryActionsPermission
+
+    .PARAMETER TeamName
+        The name of the team.
+        Note: This will be slower than querying by TeamSlug since it requires retrieving
+        all teams first.
+
+    .PARAMETER TeamSlug
+        The slug (a unique key based on the team name) of the team.
+
+    .PARAMETER AccessToken
+        If provided, this will be used as the AccessToken for authentication with the
+        REST Api.  Otherwise, will attempt to use the configured value or will run unauthenticated.
+
+    .INPUTS
+        GitHub.Branch
+        GitHub.Content
+        GitHub.Event
+        GitHub.Issue
+        GitHub.IssueComment
+        GitHub.Label
+        GitHub.Milestone
+        GitHub.Organization
+        GitHub.PullRequest
+        GitHub.Project
+        GitHub.ProjectCard
+        GitHub.ProjectColumn
+        GitHub.Reaction
+        GitHub.Release
+        GitHub.ReleaseAsset
+        GitHub.Repository
+        GitHub.Team
+
+    .OUTPUTS
+        GitHub.RepositoryTeamPermission
+
+    .EXAMPLE
+        Get-GitHubRepositoryTeamPermission -OrganizationName microsoft -Uri https://github.com/microsoft/PowerShellForGitHub -TeamName Devs
+
+        Gets permission for the Devs team on the microsoft/PowerShellForGitHub repository.
+
+    .EXAMPLE
+        Get-GitHubRepositoryTeamPermission -OrganizationName microsoft -OwnerName microsoft -RepositoryName PowerShellForGitHub -TeamName Admins
+
+        Gets permission for the Admin team on the microsoft/PowerShellForGitHub repository.
+#>
+    [CmdletBinding(DefaultParameterSetName = 'TeamNameElements')]
+    [OutputType(
+        { $script:GitHubRepositoryTeamTypeName })]
+    param
+    (
+        [Parameter(ParameterSetName = 'TeamNameElements')]
+        [Parameter(ParameterSetName = 'TeamSlugElements')]
+        [string] $OwnerName,
+
+        [Parameter(ParameterSetName = 'TeamNameElements')]
+        [Parameter(ParameterSetName = 'TeamSlugElements')]
+        [string] $RepositoryName,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamNameUri')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugUri')]
+        [Alias('RepositoryUrl')]
+        [string] $Uri,
+
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamNameElements')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugElements')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamNameUri')]
+        [Parameter(
+            Mandatory,
+            ValueFromPipelineByPropertyName,
+            ParameterSetName = 'TeamSlugUri')]
+        [ValidateNotNullOrEmpty()]
+        [string] $OrganizationName,
```

Comment:
The API has them as separate parameters: `GET /orgs/{org}/teams/{team_slug}/repos/{owner}/{repo}`, but I don't know of a scenario when they could be different. You can't add a Team to a repo that is not a member of an organization, and you can't add a team from one organization to a repo from another organization.

I'm happy to remove this parameter, or make it non-mandatory and default it to the owner name just in case there is a scenario where they can be different that we don't know about?

Reply:
If both of us are feeling pretty confident that there are no known scenarios where `OwnerName` and `OrganizationName` would be different, I'm more inclined to just remove `OrganizationName` entirely to simplify the API and make its usage a bit more obvious.  If a scenario comes up in the future where it turns out that we _do_ need it, we could add it then, but I think for now we should be shipping the API that makes the most sense for the scenarios we know about vs any hypothetical ones.

I appreciate your flexibility and understanding here in reworking the parameters and examples (in the CBH and USAGE.md) to accommodate this update.  I do think it'll simplify the API in a positive way.

Reply:
Done